jobs/haproxy/spec

```diff
@@ -406,7 +406,21 @@ properties:
         MyCustomHeader: 3
 
   ha_proxy.true_client_ip_header:
-    description: "Header to use to store the client's IP address, as seen from HAProxy. The header will be overwritten if it already exists in the request."
+    description: "Header to use to store the client's IP address, as seen from HAProxy. See forward_true_client_ip_header for more options."
+
+  ha_proxy.forward_true_client_ip_header:
+    description: |
+      This option lets you decide how to handle the header specified in ha_proxy.true_client_ip_header on any http and https frontend, when it already present in the request.
+      Your options are:
+      
+      - always_forward: Always preserve the existing header.
+      
+      - forward_only_if_route_service: This option is useful to support the header with Route Services.
+          The header will be overwritten with the current source address, unless there is an X-Cf-Proxy-Signature header.
+      
+      - always_set: Always set the header with the current source address. Will overwrite any existing header.
+
+    default: forward_only_if_route_service
```

Comment:
I know we discussed this offline, but I forgot what the reason was to have a breaking change in this property by changing the default behaviour?

Reply:
Because it's a sane default. We argued that the feature wasn't properly implemented because it just didn't work for route services at all. Now it's unified/harmonized by default.